contrib/export2collection.lua

```diff
@@ -0,0 +1,192 @@
+--[[
+
+    export2collection.lua - export a file and import it to the collection
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+--[[
+    export2collection - export an image then import it into the collection
+
+    This script provides another storage (export target) for darktable.  Selected
+    images are exported in the specified format to temporary storage.   The exported 
+    files are moved into the current collection and imported into the database.  Any 
+    tags that have been added to the original image are copied to the imported image.
+    The imported files then show up grouped with the originally selected images.
+    If there is a filename conflict on import, the incoming filename will have an 
+    increment added to it, i.e. _7D_1234.jpg would be _7D_1234_01.jpg.
+
+    USAGE
+    * require this script from your main lua file
+    * select an image or images to export
+    * in the export dialog select "file to collection" and select the format and bit depth for the
+      exported image
+    * Press "export"
+    * The exported image will be imported and grouped with the original image
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("export2collection",dt.configuration.config_dir.."/lua/locale/")
+
+local function _(msgid)
+    return gettext.dgettext("export2collection", msgid)
+end
+
+local function split_filepath(str)
```

Comment:
I actually copied and pasted from dtutils.file for the new check_if.. functions.  ;)